pkg/cluster/spec/prometheus.go

```diff
@@ -31,19 +31,21 @@ import (
 
 // PrometheusSpec represents the Prometheus Server topology specification in topology.yaml
 type PrometheusSpec struct {
-	Host            string               `yaml:"host"`
-	SSHPort         int                  `yaml:"ssh_port,omitempty" validate:"ssh_port:editable"`
-	Imported        bool                 `yaml:"imported,omitempty"`
-	Port            int                  `yaml:"port" default:"9090"`
-	DeployDir       string               `yaml:"deploy_dir,omitempty"`
-	DataDir         string               `yaml:"data_dir,omitempty"`
-	LogDir          string               `yaml:"log_dir,omitempty"`
-	NumaNode        string               `yaml:"numa_node,omitempty" validate:"numa_node:editable"`
-	Retention       string               `yaml:"storage_retention,omitempty" validate:"storage_retention:editable"`
-	ResourceControl meta.ResourceControl `yaml:"resource_control,omitempty" validate:"resource_control:editable"`
-	Arch            string               `yaml:"arch,omitempty"`
-	OS              string               `yaml:"os,omitempty"`
-	RuleDir         string               `yaml:"rule_dir,omitempty" validate:"rule_dir:editable"`
+	Host            string                       `yaml:"host"`
+	SSHPort         int                          `yaml:"ssh_port,omitempty" validate:"ssh_port:editable"`
+	Imported        bool                         `yaml:"imported,omitempty"`
+	Port            int                          `yaml:"port" default:"9090"`
+	DeployDir       string                       `yaml:"deploy_dir,omitempty"`
+	DataDir         string                       `yaml:"data_dir,omitempty"`
+	LogDir          string                       `yaml:"log_dir,omitempty"`
+	NumaNode        string                       `yaml:"numa_node,omitempty" validate:"numa_node:editable"`
+	Retention       string                       `yaml:"storage_retention,omitempty" validate:"storage_retention:editable"`
+	ResourceControl meta.ResourceControl         `yaml:"resource_control,omitempty" validate:"resource_control:editable"`
+	Arch            string                       `yaml:"arch,omitempty"`
+	OS              string                       `yaml:"os,omitempty"`
+	RuleDir         string                       `yaml:"rule_dir,omitempty" validate:"rule_dir:editable"`
+	RemoteWrite     meta.RemoteWriteServerConfig `yaml:"remote_write,omitempty" validate:"remote_write:editable"`
```

Comment:
IMHO, you should not define those two structs in the `meta` module. As `remote_write` and `remote_read` is Prometheus's spec, you can define in this file directly.

Reply:
I think you can define it like https://github.com/pingcap/tiup/blob/bc366b8941ffd94a251cfe0f7c699c6bb21fb045/pkg/cluster/spec/tiflash.go#L53-L54

use `map[string]interface{}` instead.

Reply:
fixed